### ISSUE TITLE
Some Questions about Mamba Block Training and modify

### ISSUE THREAD

**Issue by xiaosa269** [state-spaces/mamba#570]
Hello, thank you very much for your work.   I have the following issue, and I hope you can help me with it.

1.  I have no issues when using a single Mamba block. However, when I use two Mamba blocks simultaneously as shown below, after training for 30+ epochs, the loss becomes NaN.

When I stack them as shown below...
![image](https://github.com/user-attachments/assets/320bc625-fd35-47df-b20d-f8eb8209d760)
![image](https://github.com/user-attachments/assets/ac54c4e7-6c80-4f17-8772-f5f0d228968b)
![image](https://github.com/user-attachments/assets/1ba19e91-0f61-4241-87e3-acced75efa98)

...the loss becomes NaN.
![image](https://github.com/user-attachments/assets/1211c9db-1146-455e-adc9-7f5963a906e1)
![image](https://github.com/user-attachments/assets/2ed281c9-f5be-492f-8bf0-e8752a6eded9)

2.My data is point cloud data. I would like to introduce a point cloud sequence with a different sorting method into the regular Mamba to supplement the information. How should I modify the Mamba to achieve this?

Thank you very much for your help.

**Comment by XuHengYue** [comment]
> Is that pointcept codebase? Because I've met this situation and I have to set AMP=False.

yes！！I have solved this problem by adding residual connection before. Thank you for your help.

**Comment by CacatuaAlan** [comment]
> > Is that pointcept codebase? Because I've met this situation and I have to set AMP=False.
> 
> yes！！I have solved this problem by adding residual connection before. Thank you for your help.

Could you provide more detail?

**Comment by xiaosa269** [comment]
> > > Is that pointcept codebase? Because I've met this situation and I have to set AMP=False.
> > 
> > 
> > yes！！I have solved this problem by adding residual connection before. Thank you for your help.
> 
> Could you provide more detail?

for example：
 a=input
 b=mamba(input)
 output=a+b

**Comment by CacatuaAlan** [comment]
But the structure Mamba block has residual connection.

**Comment by xiaosa269** [comment]
> But the structure Mamba block has residual connection.

Yes

**Comment by CacatuaAlan** [comment]
> > But the structure Mamba block has residual connection.
> 
> Yes

It still occurs loss Nan during training.

**Comment by xiaosa269** [comment]
> > > But the structure Mamba block has residual connection.
> > 
> > 
> > Yes
> 
> It still occurs loss Nan during training.

Hello, my previous approach successfully avoided NaN issues on smaller datasets. However, when running a larger public dataset with a larger batch size, NaN issues still occur. May I ask if setting AMP=False has helped you avoid NaN problems?

**Comment by CacatuaAlan** [comment]
> > > > But the structure Mamba block has residual connection.
> > > 
> > > 
> > > Yes
> > 
> > 
> > It still occurs loss Nan during training.
> 
> Hello, my previous approach successfully avoided NaN issues on smaller datasets. However, when running a larger public dataset with a larger batch size, NaN issues still occur. May I ask if setting AMP=False has helped you avoid NaN problems?

Yes! But it turns out a large consumption.

**Comment by xiaosa269** [comment]
> > > > > But the structure Mamba block has residual connection.
> > > > 
> > > > 
> > > > Yes
> > > 
> > > 
> > > It still occurs loss Nan during training.
> > 
> > 
> > Hello, my previous approach successfully avoided NaN issues on smaller datasets. However, when running a larger public dataset with a larger batch size, NaN issues still occur. May I ask if setting AMP=False has helped you avoid NaN problems?
> 
> Yes! But it turns out a large consumption.

Yes, I also noticed the issue of high memory consumption. While reviewing the issues in Pointcept, I came across this one (https://github.com/Pointcept/Pointcept/issues/283). Have you tried using the method mentioned there: torch.nn.utils.clip_grad_norm_(self.model.parameters(), max_norm=1.0)?

**Comment by CacatuaAlan** [comment]
> > > > > > But the structure Mamba block has residual connection.
> > > > > 
> > > > > 
> > > > > Yes
> > > > 
> > > > 
> > > > It still occurs loss Nan during training.
> > > 
> > > 
> > > Hello, my previous approach successfully avoided NaN issues on smaller datasets. However, when running a larger public dataset with a larger batch size, NaN issues still occur. May I ask if setting AMP=False has helped you avoid NaN problems?
> > 
> > 
> > Yes! But it turns out a large consumption.
> 
> Yes, I also noticed the issue of high memory consumption. While reviewing the issues in Pointcept, I came across this one ([Pointcept/Pointcept#283](https://github.com/Pointcept/Pointcept/issues/283)). Have you tried using the method mentioned there: torch.nn.utils.clip_grad_norm_(self.model.parameters(), max_norm=1.0)?

Pointcept has integrated the clip_grad_norm and I tried it before. I still get loss Nan :(

**Comment by xiaosa269** [comment]
> > > > > > > But the structure Mamba block has residual connection.
> > > > > > 
> > > > > > 
> > > > > > Yes
> > > > > 
> > > > > 
> > > > > It still occurs loss Nan during training.
> > > > 
> > > > 
> > > > Hello, my previous approach successfully avoided NaN issues on smaller datasets. However, when running a larger public dataset with a larger batch size, NaN issues still occur. May I ask if setting AMP=False has helped you avoid NaN problems?
> > > 
> > > 
> > > Yes! But it turns out a large consumption.
> > 
> > 
> > Yes, I also noticed the issue of high memory consumption. While reviewing the issues in Pointcept, I came across this one ([Pointcept/Pointcept#283](https://github.com/Pointcept/Pointcept/issues/283)). Have you tried using the method mentioned there: torch.nn.utils.clip_grad_norm_(self.model.parameters(), max_norm=1.0)?
> 
> Pointcept has integrated the clip_grad_norm and I tried it before. I still get loss Nan :(

That's truly a disappointing piece of news. Thank you for your response.
